### ISSUE TITLE
Previous node version is causing side effects.

### ISSUE THREAD

**Issue by smussell** [tj/n#236]
Under some circumstances n is causing my npm install from github to fail.  I've tried on my machine and also a colleagues machine so it is reproducible.  This is happening under Mac os 10.10.3 I have the following package.json

```
{
  "name": "test",
  "description": "test",
  "version": "0.0.1",
  "private": true,
  "keywords": "",
  "dependencies": {
    "react-grid-layout": "git+https://github.com/STRML/react-grid-layout.git"

  }
}
```

and I see the following error:

```
npm ERR! fetch failed https://github.com/strml/react-draggable
npm WARN retry will retry, error on last attempt: Error: fetch failed with status code 406
npm ERR! fetch failed https://github.com/strml/react-draggable
npm WARN retry will retry, error on last attempt: Error: fetch failed with status code 406
npm ERR! fetch failed https://github.com/strml/react-draggable
npm ERR! Darwin 14.3.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v0.12.0
npm ERR! npm  v2.5.1

npm ERR! fetch failed with status code 406
```

So my steps to reproduce are:
1. brew install node
2. npm install // works fine
3. npm install -g n
4. n 0.10.26
5. npm install // works fine
6. n 0.12.0
7. npm install // I see the above error

What's strange is that if I replace step 4 with n 0.10.36 all is fine. Something about having previously installed .26 is causing the issue, it seems then that something is not being cleaned up properly when switching versions.

**Comment by maxrimue** [comment]
You're right, for some reason, the `https` doesn't work when I test it with node 0.12.0 as well, it does work however when I change the dependency to `"react-grid-layout": "STRML/react-grid-layout"`. I'm not sure if this really is a problem of `n` however when just certain URL strings don't work.

**Comment by smussell** [comment]
The only reason I believe it to be an issue with n is that everything works until I use n to install 0.10.26 and then switch back to 0.12.  That makes me believe n is leaving something behind that is causing side effects across node versions.  Thanks for looking at it.

**Comment by maxrimue** [comment]
@smussell I get the feeling that this problem is more connected with npm, and there's probably something working wrong when overwriting the npm installation when you change from this specific version to 0.12.0. You could try to update npm via npm, that worked for me. n however isn't capable of isolating the npm installation which means you have to update npm everytime you do a version change with n.

**Comment by shadowspawn** [comment]
This was probably fixed by #470 

This issue has not had any activity in over six months. It isn't likely to get acted on due to this report.

Feel free to open a new issue if it comes up again, with new information and renewed interest.
 
Thank you  for your contributions.
